src/types/common/interfaces/common.interfaces.ts

```diff
@@ -94,3 +94,68 @@ export interface FormData {
   password: string
   confirmPassword: string
 }
+
+export interface Media {
+  name: string
+  path: string
+}
+
+export interface File extends CommonEntityFields {
+  name: string
+  size: number
+  url?: string
+}
+
+export interface Link {
+  name: string
+  url: string
+}
+
+export interface Lesson extends CommonEntityFields {
+  title: string
+  category?: {
+    name: string
+  }
+}
+
+export interface Quiz extends CommonEntityFields {
+  title: string
+  category?: {
+    name: string
+  }
+}
+
+export interface CourseResources {
+  _id: string
+  title: string
+  category?: { _id: string; name: string }
+  createdAt: Date
+  updatedAt: Date
+}
+
+export interface Question {
+  _id: string
+  title: string
+  text: string
+  answers: Answer[]
+  author: string
+  type: QuestionTypesEnum
+  category?: Category | null
+  createdAt: string | Date
+  updatedAt: string | Date
```

Comment:
Agree with the bot. In `CourseResources` we used a simple `Date` type in the same field